### ISSUE TITLE
[Bug] 跑tade项目loss一直在0.3左右，iou一直都在0.014左右

### ISSUE THREAD

**Issue by peach-gif** [happytianhao/TaDe#2]
### Prerequisite

- [x] I have searched [Issues](https://github.com/open-mmlab/mmdetection3d/issues) and [Discussions](https://github.com/open-mmlab/mmdetection3d/discussions) but cannot get the expected help.
- [x] I have read the [FAQ documentation](https://mmdetection3d.readthedocs.io/en/latest/notes/faq.html) but cannot get the expected help.
- [x] The bug has not been fixed in the [latest version (dev-1.x)](https://github.com/open-mmlab/mmdetection3d/tree/dev-1.x) or [latest version (dev-1.0)](https://github.com/open-mmlab/mmdetection3d/tree/dev-1.0).

### Task

I'm using the official example scripts/configs for the officially supported tasks/models/datasets.

### Branch

main branch https://github.com/open-mmlab/mmdetection3d

### Environment

PyTorch: 2.1.0+cu121
Torchvision: 0.16.0+cu121
CUDA available: True
CUDA version: 12.1
Python 3.8.5

### Reproduces the problem - code sample

mean/iou@0.60: 0.0158  mIOU: 0.0453  data_time: 0.0058  time: 0.1100  loss：0.3左右

### Reproduces the problem - command or script

import torch
import torch.nn as nn
import torch.nn.functional as F
import numpy as np
from mmengine.registry import MODELS
from mmengine.model import BaseModel
from .map_converter import MapConverter
from .utils import visualize

@MODELS.register_module()
class BEV(BaseModel):
    def __init__(
        self,
        backbone,
        neck,
        n_class=14,
        dim=256,
        h_polar=256,
        w_polar=704,
        class_weights=[],
        stage=0,
        vis_list=[],
        init_cfg=None,
    ):
        super().__init__(init_cfg=init_cfg)
        self.backbone = MODELS.build(backbone)
        self.neck = MODELS.build(neck)
        self.pos_src = nn.Parameter(torch.randn(1, dim, 64, 1))
        self.pos_tgt = nn.Parameter(torch.randn(1, dim, 64, 1))
        self.transformer = nn.Transformer(
            d_model=dim,
            num_encoder_layers=2,
            num_decoder_layers=2,
            dim_feedforward=dim,
        )
        self.cnn = nn.Sequential(
            nn.Conv2d(dim, dim, 4, 2, 1),
            nn.BatchNorm2d(dim),
            nn.ReLU(True),
            nn.Conv2d(dim, dim, 4, 2, 1),
            nn.BatchNorm2d(dim),
            nn.ReLU(True),
            nn.Conv2d(dim, dim, 4, 2, 1),
            nn.BatchNorm2d(dim),
        )
        self.encoder = nn.Sequential(
            nn.Conv2d(n_class, dim, 4, 2, 1),
            nn.BatchNorm2d(dim),
            nn.ReLU(True),
            nn.Conv2d(dim, dim, 4, 2, 1),
            nn.BatchNorm2d(dim),
            nn.ReLU(True),
            nn.Conv2d(dim, dim, 4, 2, 1),
            nn.BatchNorm2d(dim),
            nn.ReLU(True),
            nn.Conv2d(dim, dim, 4, 2, 1),
            nn.BatchNorm2d(dim),
            nn.ReLU(True),
            nn.Conv2d(dim, dim, 4, 2, 1),
            nn.BatchNorm2d(dim),
        )
        self.decoder = nn.Sequential(
            nn.ConvTranspose2d(dim, dim, 4, 2, 1),
            nn.BatchNorm2d(dim),
            nn.ReLU(True),
            nn.ConvTranspose2d(dim, dim, 4, 2, 1),
            nn.BatchNorm2d(dim),
            nn.ReLU(True),
            nn.ConvTranspose2d(dim, dim, 4, 2, 1),
            nn.BatchNorm2d(dim),
            nn.ReLU(True),
            nn.ConvTranspose2d(dim, dim, 4, 2, 1),
            nn.BatchNorm2d(dim),
            nn.ReLU(True),
            nn.ConvTranspose2d(dim, n_class, 4, 2, 1),
            # nn.Sigmoid(),
        )

        self.h_polar, self.w_polar = h_polar, w_polar
        self.map_converter_infos, self.map_converters = [], []
        self.class_weights = torch.tensor(class_weights)
        assert stage in [0, 1, 2, 3]
        self.stage = stage
        self.vis_list = vis_list
        self.epoch = None

    def init_map_converters(self, data_samples, device):
        map_converter_indices = []
        for data_sample in data_samples:
            for cam_item in data_sample.cams:
                map_converter_info = dict(
                    h_polar=self.h_polar,
                    w_polar=self.w_polar,
                    h_cart=cam_item["bev_shape"][1],
                    w_cart=cam_item["bev_shape"][2],
                    extents_polar=[0, 53],
                    extents_cart=cam_item["extents"],
                    cam2img=cam_item["cam2img"],
                    image_width=cam_item["image_width"],
                    colormap=cam_item["colormap"],
                    device=device,
                )
                if map_converter_info not in self.map_converter_infos:
                    self.map_converter_infos.append(map_converter_info)
                    self.map_converters.append(MapConverter(**map_converter_info))
                map_converter_indices.append(self.map_converter_infos.index(map_converter_info))
        return map_converter_indices

    def cart2polar(self, map_cart, map_converter_indices):
        """Convert map from cart to polar.

        Args:
            map_cart (torch.bool): (N, *, 200, 200)

        Returns:
            map_polar (torch.bool): (N, *, 256, 352)
        """
        map_polar = []
        for i, map_converter_index in enumerate(map_converter_indices):
            map_polar.append(self.map_converters[map_converter_index].cart2polar(map_cart[i]))
        return torch.stack(map_polar)

    def polar2cart(self, map_polar, map_converter_indices):
        """Convert map from polar to cart.

        Args:
            map_polar (torch.float): (N, *, 256, 352)

        Returns:
            map_cart (torch.float): (N, *, 200, 200)
        """
        map_cart = []
        for i, map_converter_index in enumerate(map_converter_indices):
            map_cart.append(self.map_converters[map_converter_index].polar2cart(map_polar[i]))
        return torch.stack(map_cart)

    def forward(self, data_samples, inputs, mode=None):
        image = torch.concat(inputs["image"])
        bev_map = torch.concat(inputs["bev_map"])
        map_converter_indices = self.init_map_converters(data_samples, bev_map.device)
        expand_map = self.cart2polar(bev_map, map_converter_indices)

        x = expand_map[:, :-1].float()

        if mode == "loss":
            if np.random.rand() < 0.5:
                image = torch.flip(image, [3])
                x = torch.flip(x, [3])
            if self.stage in [0, 2, 3]:
                feature = self.neck(self.backbone(image))
                feature = torch.flip(feature[0], [2])

                n, c, h, w = feature.size()
                    # --- 第一行加在这里：输入标准化 ---
                # 作用：将 Backbone 输出的特征值强制拉回到均值 0, 方差 1 附近，
                # 防止由于特征值过大导致 Transformer 的 Attention 权重计算溢出。
                feature = (feature - feature.mean()) / (feature.std() + 1e-6)
                src, tgt = feature + self.pos_src, torch.zeros_like(feature) + self.pos_tgt
                src, tgt = src.permute(2, 0, 3, 1).reshape(h, -1, c), tgt.permute(2, 0, 3, 1).reshape(h, -1, c)
                # --- 第二行加在这里：包裹 Transformer 调用 ---
                # 作用：强制 PyTorch 使用最稳健的数学公式计算，跳过报错的 Flash Attention 加速算子。
                with torch.backends.cuda.sdp_kernel(enable_flash=False, enable_math=True, enable_mem_efficient=False):
                    feature = self.transformer(src, tgt)
                
                feature = feature.reshape(h, n, w, c).permute(1, 3, 0, 2)

                feature = self.cnn(feature)
            if self.stage in [1, 2]:
                ze = self.encoder(x)  # (N, 14, 256, 704) -> (N, 256, 8, 22)

            if self.stage == 0:
                x_recon = self.decoder(feature) # 注意：这里现在是 Logits（未经过 Sigmoid）
                # 修正：将 weight 改为 pos_weight
                loss = F.binary_cross_entropy_with_logits(
                    x_recon, 
                    x, 
                    pos_weight=self.class_weights.to(x).view(1, -1, 1, 1)
                )
            
            elif self.stage == 1:
                noise = torch.randn_like(ze)
                x_recon = self.decoder(np.sqrt(0.5) * ze + np.sqrt(0.5) * noise)
                # 修正：stage 1 也必须改用 with_logits，因为 decoder 已经没 Sigmoid 了
                loss = F.binary_cross_entropy_with_logits(
                    x_recon, 
                    x, 
                    pos_weight=self.class_weights.to(x).view(1, -1, 1, 1)
                )

            elif self.stage == 2:
                loss = F.mse_loss(feature, ze.detach())

            elif self.stage == 3:
                x_recon = self.decoder(feature.detach())
                # 修正：stage 3 也同步修改
                loss = F.binary_cross_entropy_with_logits(
                    x_recon, 
                    x, 
                    pos_weight=self.class_weights.to(x).view(1, -1, 1, 1)
                )
            return dict(loss=loss)
        else:
            if self.stage in [0, 2, 3]:
                with torch.no_grad():
                    feature = self.neck(self.backbone(image))
                    feature = torch.flip(feature[0], [2])
    # --- 第一行加在这里：输入标准化 ---
                # 作用：将 Backbone 输出的特征值强制拉回到均值 0, 方差 1 附近，
                # 防止由于特征值过大导致 Transformer 的 Attention 权重计算溢出。
                    feature = (feature - feature.mean()) / (feature.std() + 1e-6)
                    n, c, h, w = feature.size()
                
                    src, tgt = feature + self.pos_src, torch.zeros_like(feature) + self.pos_tgt
                    src, tgt = src.permute(2, 0, 3, 1).reshape(h, -1, c), tgt.permute(2, 0, 3, 1).reshape(h, -1, c)
                    # --- 第二行加在这里：包裹 Transformer 调用 ---
                # 作用：强制 PyTorch 使用最稳健的数学公式计算，跳过报错的 Flash Attention 加速算子。
                    with torch.backends.cuda.sdp_kernel(enable_flash=False, enable_math=True, enable_mem_efficient=False):
                        feature = self.transformer(src, tgt)
                    feature = feature.reshape(h, n, w, c).permute(1, 3, 0, 2)

                    feature = self.cnn(feature)
            if self.stage == 1:
                with torch.no_grad():
                    feature = self.encoder(x)  # (N, 14, 256, 704) -> (N, 256, 8, 22)
            with torch.no_grad():
                logits = self.decoder(feature)
                x_recon = torch.sigmoid(logits) # 这里必须手动转回 0-1 概率

            x_bev = self.polar2cart(x_recon, map_converter_indices)
            for i, data_sample in enumerate(data_samples):
                for j, cam in enumerate(data_sample.cams):
                    if cam["image_path"] in self.vis_list:
                        folder_name = cam["camera_type"] + "_" + cam["image_path"].split("/")[-1].split(".")[0]
                        visualize(
                            f"visualizations/stage_{self.stage}/epoch_{self.epoch}/{folder_name}",
                            bev_map=bev_map[i * len(data_sample.cams) + j],
                            expand_map=expand_map[i * len(data_sample.cams) + j],
                            x_recon=x_recon[i * len(data_sample.cams) + j] >= 0.5,
                            x_bev=x_bev[i * len(data_sample.cams) + j] >= 0.5,
                            image=cam["image_path"],
                        )

            return x_bev, bev_map

### Reproduces the problem - error message

mean/iou@0.60: 0.0158  mIOU: 0.0453  data_time: 0.0058  time: 0.1100
